File: generate_icons.py
#!/usr/bin/env python3
"""
Generer PNG-ikoner fra SVG for SykePriser.app

Krav: pip install cairosvg pillow

Kjør: python generate_icons.py
"""

import os
import sys

try:
    import cairosvg
    from PIL import Image
    import io
except ImportError:
    print("❌ Mangler avhengigheter!")
    print("\nInstaller med:")
    print("  pip install cairosvg pillow")
    print("\nEller bruk online verktøy:")
    print("  https://cloudconvert.com/svg-to-png")
    sys.exit(1)

def svg_to_png(svg_path, png_path, size):
    """Konverter SVG til PNG med gitt størrelse"""
    # Les SVG
    with open(svg_path, 'rb') as f:
        svg_data = f.read()

    # Konverter SVG til PNG bytes
    png_data = cairosvg.svg2png(
        bytestring=svg_data,
        output_width=size,
        output_height=size
    )

    # Lagre som PNG
    img = Image.open(io.BytesIO(png_data))
    img.save(png_path, 'PNG')
    print(f"✅ Genererte: {png_path} ({size}x{size})")

def main():
    script_dir = os.path.dirname(os.path.abspath(__file__))
    public_dir = os.path.join(script_dir, 'public')
    svg_path = os.path.join(public_dir, 'icon.svg')

    if not os.path.exists(svg_path):
        print(f"❌ Finner ikke {svg_path}")
        sys.exit(1)

    # Generer ikoner
    svg_to_png(svg_path, os.path.join(public_dir, 'icon.png'), 192)
    svg_to_png(svg_path, os.path.join(public_dir, 'icon-512.png'), 512)

    print("\n🎉 Ikoner generert! Klar til deploy.")

if __name__ == '__main__':
    main()

File: generate_icons.js
#!/usr/bin/env node
/**
 * Generer PNG-ikoner fra SVG for SykePriser.app
 *
 * Krav: npm install sharp
 *
 * Kjør: node generate_icons.js
 */

const fs = require('fs');
const path = require('path');

let sharp;
try {
    sharp = require('sharp');
} catch (error) {
    console.error('❌ Mangler avhengigheter!');
    console.error('\nInstaller med:');
    console.error('  npm install sharp');
    console.error('\nEller bruk online verktøy:');
    console.error('  https://cloudconvert.com/svg-to-png');
    process.exit(1);
}

async function svgToPng(svgPath, pngPath, size) {
    try {
        await sharp(svgPath)
            .resize(size, size)
            .png()
            .toFile(pngPath);
        console.log(`✅ Genererte: ${pngPath} (${size}x${size})`);
    } catch (error) {
        console.error(`❌ Feil ved generering av ${pngPath}:`, error.message);
        throw error;
    }
}

async function main() {
    const publicDir = path.join(__dirname, 'public');
    const svgPath = path.join(publicDir, 'icon.svg');

    if (!fs.existsSync(svgPath)) {
        console.error(`❌ Finner ikke ${svgPath}`);
        process.exit(1);
    }

    // Generer ikoner
    await svgToPng(svgPath, path.join(publicDir, 'icon.png'), 192);
    await svgToPng(svgPath, path.join(publicDir, 'icon-512.png'), 512);

    console.log('\n🎉 Ikoner generert! Klar til deploy.');
}

main().catch(error => {
    console.error('❌ Feil:', error);
    process.exit(1);
});
